### ISSUE TITLE
WP 3.5.2 D Prepare SLURM queuing machinery for all sky pipeline runs. 

### ISSUE THREAD

**Issue by tms-epcc** [lsst-uk/lsst-ir-fusion#33]

**Comment by tms-epcc** [comment]
As reported at the WP3.5 meeting on [06/SEP/21](https://lsst-uk.atlassian.net/wiki/download/attachments/766869533/LUSC-WP3.5-PM-meeting-2021-09-06-Minutes.txt?api=v2), there has been no progress on this since the gen 3 butler work  has been the priority.  Once gen 3 butler is ready, then this will	 be tackled.   Subsequently, at the WP3.5 meeting on 23/SEP/21, it was reported that it had been decided to postpone. WP3.5 have an 800 sq-deg run completed and don't think there will be a need for an all-sky run till much later in the project.

**Comment by raphaelshirley** [comment]
We have developed a SLURM machinery using large job arrays with one job per image for the early stages and one job per patch after the coadding stages. This is now all in place. Example SLURM scripts for the full VIKING HSC PDR 2 overlap are available here:

https://github.com/lsst-uk/lsst-ir-fusion/tree/master/dmu4/dmu4_VIKING/slurm

That directory contains the job dictionaries in json format. These dictionaries convert an integer job array number to the job details telling the pipeline which image, patch, band etc to run on. The scripts which run the gen 2 pipelines and the SLURM scripts numbered in the order with which they need to be submitted. The only difference moving to gen 3 will be changing the pipeline scripts; the SLURM scripts and job dictionaries will be the same.
